lib/cog/chat/hipchat/template_processor.ex

```diff
@@ -31,9 +31,20 @@ defmodule Cog.Chat.HipChat.TemplateProcessor do
     do: "<code>#{text}</code>"
   defp process_directive(%{"name" => "fixed_width_block", "text" => text}),
     do: "<pre>#{text}</pre>"
-  defp process_directive(%{"name" => "paragraph", "children" => children}) do
-      Enum.map_join(children, &process_directive/1) <> "<br/><br/>"
-  end
+  defp process_directive(%{"name" => "paragraph", "children" => children}),
+    do: Enum.map_join(children, &process_directive/1) <> "<br/><br/>"
+  # If you try to render a link with nil or blank text, HipChat displays nothing to the user.
+  # This way at least a link is rendered. This basically mirrors the default Slack behavior.
+  defp process_directive(%{"name" => "link", "text" => text, "url" => url}) when text in [nil, ""],
+    do: "<a href='#{url}'>#{url}</a>"
+  # Rendering a link in HipChat with a nil or blank url will obviously result in an invalid link. We
+  # inform the user inline that there was a problem and log a warning.
+  defp process_directive(%{"name" => "link", "text" => text, "url" => url}=directive) when url in [nil, ""] do
+    Logger.warn("Invalid link; #{inspect directive}")
+    ~s[(invalid link! text:"#{text}" url: "#{url}")]
```

Comment:
You should do `#{inspect blah}` here so you can tell the difference between nil and `""`